samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -157,7 +178,41 @@ static int setup_raw_pkt_socket(int *sockfd, struct sockaddr_ll *sa)
 	return 0;
 }
 
-static void fill_raw_tx_pkt_hdr(struct raw_tx_pkt_header *raw_tx_pkt)
+int validate(int value, int min, int max, const char *param)
+{
+	if (value < min || value > max) {
+		LOG_ERR("Please provide %s value between %d and %d", param, min, max);
+		return -1;
+	}
+
+	return 0;
+}
+
+int validate_rate(int data_rate, int flag)
+{
+	if ((flag == 0) && ((data_rate == 1) ||
```

Comment:
Can we have enum defined atleast for the `flag` instead of using hard coded values ?